module/documents/activity/order.mjs

```diff
@@ -164,10 +164,13 @@ export default class OrderActivity extends ActivityMixin(OrderActivityData) {
       }
       updates["system.trade.pending.value"] = trade.sell ? (trade.creatures.price ?? 0) : costs.gold;
       updates["system.trade.pending.creatures"] = creatures;
-      // TODO: We need a way to visualize 'pending' animal purchases/sales. For now update immediately.
-      updates["system.trade.creatures.value"] = trade.sell
-        ? system.trade.creatures.value.filter((_, i) => !trade.creatures.sell[i])
-        : system.trade.creatures.value.concat(creatures);
+
+      // Sold livestock are removed immediately. Bought livestock remain pending until the order is complete.
+      if ( trade.sell ) {
+        updates["system.trade.creatures.value"] = system.trade.creatures.value.filter((_, i) => {
+          return !trade.creatures.sell[i];
+        });
+      }
     }
   }
```

Comment:
This change looks good, but I noticed in testing that line `273` throws a deprecation warning in V12 due to passing `fromUuid` directly into `map`.

```javascript
else creatures.push(...await Promise.all(trade.creatures.buy.filter(_ => _).map(u => fromUuid(u))));
```